cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSQLEngineAdapter.java

```diff
@@ -170,7 +177,20 @@ public SQLDataset pushInternal(String datasetName,
     // Create push request
     SQLPushRequest pushRequest = new SQLPushRequest(datasetName, schema);
 
-    //Get the push provider and wait for it to be ready to use
+    // Check if any of the declared capabilities for this plugin is able to consume this Push Request.
+    // If so, we will process this request using a consumer.
+    for (PushCapability capability : sqlEngine.getPushCapabilities()) {
+      SQLDatasetConsumer<StructuredRecord> consumer = sqlEngine.getConsumer(pushRequest, capability);
+
+      // If a consumer is able to consume this request, we delegate the execution to the consumer.
+      if (consumer != null) {
+        RecordCollection<StructuredRecord> recordCollection =
+          new SparkRecordCollectionImpl<>(collection.getUnderlying());
+        return consumer.consume(recordCollection);
```

Comment:
As discussed, can we switch to `Dataframe` instead of `RDD<StructuredRecord>` here?

Reply:
Using `Dataframe` instead of `RDD` would imply much larger changes to the `RDDCollection` class, as this is all being invoked from there.

I can make this changes in this PR or I can open a later PR for this (the latter would be my preference)

Reply:
I mean the BatchSQLEngineAdapter can still convert Dataframe to RDD. But I don't think we should put the burden of handling StructuredRecords to SQLEngine. After all, StructuredRecord may be not needed at all going forward. If you want to do this change separately, it's possible, but it would be a prerequisite to this PR. I don't want us to go forward with more implementation before this change is done

Reply:
TLDR: At this point it's more about Platform<>SQLEngine contract, not Platform internals.

Reply:
Updated. PTAL